exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/DatabaseConfig.kt

```diff
@@ -124,8 +129,17 @@ class DatabaseConfig private constructor(
                 keepLoadedReferencesOutOfTransaction = builder.keepLoadedReferencesOutOfTransaction,
                 explicitDialect = builder.explicitDialect,
                 defaultSchema = builder.defaultSchema,
-                logTooMuchResultSetsThreshold = builder.logTooMuchResultSetsThreshold
+                logTooMuchResultSetsThreshold = builder.logTooMuchResultSetsThreshold,
+                preserveKeywordCasing = builder.preserveKeywordCasing,
             )
         }
     }
 }
+
+@RequiresOptIn(
+    message = "This API is experimental and the behavior defined by setting this value to 'true' may become the default " +
+        "in future releases. Its usage must be marked with '@OptIn(org.jetbrains.exposed.sql.ExperimentalKeywordApi::class)' " +
+        "or '@org.jetbrains.exposed.sql.ExperimentalKeywordApi'."
+)
+@Target(AnnotationTarget.PROPERTY)
+annotation class ExperimentalKeywordApi
```

Comment:
`preserveKeywordCasing` is now set to false by default and requires opt-in to set to true.

I tried to follow naming and message conventions from Kotlin and Compose Multiplatform repos.